sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/XmlSensor.java

```diff
@@ -52,11 +60,21 @@
  */
 public class XmlSensor implements Sensor {
 
+  /**
+   * Use Sonar logger instead of SL4FJ logger, in order to be able to unit test the logs.
+   */
+  private static final Logger LOG = Loggers.get(XmlSensor.class);
+
+  private static final Version V6_0 = Version.create(6, 0);
+
   private final Checks<Object> checks;
   private final FileSystem fileSystem;
   private final FilePredicate mainFilesPredicate;
   private final FileLinesContextFactory fileLinesContextFactory;
 
+  // parsingErrorRuleKey is null if ParsingErrorCheck is not activated
+  private RuleKey parsingErrorRuleKey = null;
```

Comment:
@pynicolas Agreed. I had the same feeling when I wrote that code: calling method initParsingErrorKey() in the top-level method execute() is ugly. But the current implementation avoids the recalculation of parsingErrorKey at every invocation of processParseException(). Any suggestion?